python/dgl/graphbolt/internal_utils.py

```diff
@@ -18,6 +19,22 @@
 _default_formatwarning = warnings.formatwarning
 
 
+def is_cuda_available():
+    try:
+        # This op is defined if graphbolt is built with CUDA support.
+        _ = torch.ops.graphbolt.set_max_uva_threads
+        graphbolt_cuda_available = True
+    except AttributeError:
+        graphbolt_cuda_available = False
+
+    if graphbolt_cuda_available != torch.cuda.is_available():
+        _boolean_to_enabled_disabled = {True: "enabled", False: "disabled"}
+        gb_warning(
```

Comment:
Should we make this an error instead?

Reply:
Why? GB works without CUDA support, right?

Reply:
It does. However, in gb.DataLoader, we rely on `torch.cuda.is_available()` so the users encounter the issues posted in the PR description.

Reply:
Yes, we should add is_available check in the corresponding code and throw error.

Reply:
What do you mean exactly?

Reply:
Sorry, I misunderstood, this is an internal util. Yes, feel free to throw error.
If this is an external util to for user to check whether is_cuda_available in this package, then throw warning only.

Reply:
So should we not let users who installed CPU version of GraphBolt when CUDA is available at all by throwing an error during initialization of GB library?

Reply:
I think the users should install the right version of the library, if they have a GPU but install the CPU version, then they won't have the best experience using our library. After merging this PR, I will open another one turning the warning into an error.

Reply:
See #7514.